.buildkite/pull_request_pipeline.yml

```diff
@@ -46,18 +48,31 @@ steps:
       automatic:
         - limit: 3
     command: |
-      set -euo pipefail
-      if [[ $BUILDKITE_PULL_REQUEST == "false" ]]; then
-        # https://github.com/elastic/logstash/pull/15486 for background
-        export ENABLE_SONARQUBE="false"
-      else
-        source .buildkite/scripts/pull-requests/sonar-env.sh
-      fi
-
+      set -euo pipefail      
       source .buildkite/scripts/common/container-agent.sh
       ci/unit_tests.sh java
     artifact_paths:
       - "**/build/test-results/javaTests/TEST-*.xml"
+      - "**/jacocoTestReport.xml"
+      - "**/build/classes/**/*.*"
+
+  - label: ":sonarqube: Continuous Code Inspection"
+    env:
+      VAULT_SONAR_TOKEN_PATH: "kv/ci-shared/platform-ingest/elastic/logstash/sonar-analyze-token"
+    agents:
+      image: "docker.elastic.co/cloud-ci/sonarqube/buildkite-scanner:latest"
+    command:
+      - "buildkite-agent artifact download --step ruby-unit-tests coverage/coverage.json ."
+      - "buildkite-agent artifact download --step java-unit-tests **/jacocoTestReport.xml ."
+      - "buildkite-agent artifact download --step java-unit-tests **/build/classes/**/*.* ."
+      - "ls -lah"
+      - "/scan-source-code.sh"
```

Comment:
The pull_request pipeline is also invoked by the first phase of the exhaustive tests (see [here](https://github.com/elastic/logstash/blob/16c4d8827e0dee9345d5e3cb8d3580bc463a6c0b/.buildkite/scripts/exhaustive-tests/generate-steps.py#L29) and [here](https://buildkite.com/elastic/logstash-exhaustive-tests-pipeline/builds/101#018cfe60-0913-4ada-a561-b66023925e67/96-99) for the resulting steps.

**However** we only want to run code inspection in PRs, not when this is used via other pipelines. You've probably seen this done [above](https://github.com/elastic/logstash/pull/15810/files#diff-044e110c951f71fa4ff9db56f0821bac0de6e1d091d779b448bd6f14e1ad1c0dL50) using the `$BUILDKITE_PULL_REQUEST` env var. So I recommend that we leverage [Buildkite step filters](https://buildkite.com/docs/pipelines/conditionals#conditionals-in-steps) and [variables](https://buildkite.com/docs/pipelines/conditionals#variable-and-syntax-reference-variables) to only run this step when triggered by pull requests by adding something like:

```
if: "build.pull_request.id != null"
```